components/brave_wallet/browser/zcash/rust/lib.rs

```diff
@@ -411,3 +480,90 @@ impl OrchardUnauthorizedBundle {
     }
 }
 
+impl ShieldedOutput<OrchardDomain, COMPACT_NOTE_SIZE> for OrchardCompactAction {
+    fn ephemeral_key(&self) -> EphemeralKeyBytes {
+        EphemeralKeyBytes(self.ephemeral_key)
+    }
+
+    fn cmstar_bytes(&self) -> [u8; 32] {
+        self.cmx
+    }
+
+    fn enc_ciphertext(&self) -> &[u8; COMPACT_NOTE_SIZE] {
+        &self.enc_cipher_text
+    }
+}
+
+fn batch_decode(
```

Comment:
Yes, we have tests for OrchardBlockDecoderImpl where it is used

Reply:
I'm not seeing any tests in components/brave_wallet/browser/internal/orchard_block_scanner_unittest.cc where we're checking for decoding errors and making sure that's being handled properly. Could we add some there since that's actually a good place to check for higher level error handling as we cross the FFI between C++ and Rust.

Reply:
There is  OrchardBlockScannerTest::WrongInput. I've added additional cases when cmx\nullifier data doesn't match bls\pasta restrictions